src/BackgroundTask/CronBasedBackgroundTask.php

```diff
@@ -20,20 +18,20 @@
  * @since 2.0
  * @internal
  */
-abstract class CronBasedBackgroundTask implements Service, Registerable, Conditional, Deactivateable {
+abstract class CronBasedBackgroundTask implements Service, Registerable, Conditional {
 
 	const DEFAULT_INTERVAL_HOURLY      = 'hourly';
 	const DEFAULT_INTERVAL_TWICE_DAILY = 'twicedaily';
 	const DEFAULT_INTERVAL_DAILY       = 'daily';
 
 	/**
-	 * Name of the plugin as WordPress is expecting it.
+	 * Class constructor.
 	 *
-	 * This should usually have the form "amp/amp.php".
-	 *
-	 * @var string
+	 * @param BackgroundTaskDeactivator $background_task_deactivator Service that deactivates background events.
 	 */
-	private $plugin_file;
+	public function __construct( BackgroundTaskDeactivator $background_task_deactivator ) {
+		$background_task_deactivator->add_event( $this->get_event_name() );
```

Comment:
Would this be better suited to be called in the `register` method?

Reply:
Good call. Done in 5334bdf62